File: bilan_sky/bilan_air_booking_system/api/passenger.py
# bilan_air/api/passenger.py

import frappe

@frappe.whitelist(allow_guest=True)
def register_passenger(passenger_data):
    """Create a new passenger"""
    
    existing = frappe.db.exists("Passenger", {"id_number": passenger_data.get("id_number")})
    
    if existing:
        return frappe.get_doc("Passenger", existing)
    
    passenger = frappe.get_doc({
        "doctype": "Passenger",
        "full_name": passenger_data.get("full_name"),
        "passenger_type": passenger_data.get("passenger_type", "Adult"),
        "id_number": passenger_data.get("id_number"),
        "date_of_birth": passenger_data.get("date_of_birth"),
        "phone_number": passenger_data.get("phone_number"),
        "email": passenger_data.get("email"),
        "nationality": passenger_data.get("nationality"),
        "notes": passenger_data.get("notes"),
        "is_active": passenger_data.get("is_active", 1),
    })
    passenger.insert()
    frappe.db.commit()
    
    return passenger

@frappe.whitelist(allow_guest=True)
def lookup_passenger_by_id(id_number):
    """Get passenger by ID number"""
    
    passengers = frappe.get_all("Passenger",
        filters={"id_number": id_number},
        fields=["name", "full_name", "passenger_type", "id_number", "phone_number", "email"]
    )
    
    return passengers[0] if passengers else None

File: frontend/services/passenger.ts
import { apiRequest, methodUrl } from "./apiClient";
import type { PaginatedResponse } from "@/types/bilan";

export interface PassengerData {
  full_name: string;
  id_number?: string;
  date_of_birth: string;
  passenger_type: string;
  phone_number: string;
  email: string;
  nationality?: string;
  notes?: string;
  is_active?: number;
}

export interface PassengerRecord extends PassengerData {
  name: string;
  is_active?: number;
  notes?: string;
}

export async function registerPassenger(passenger_data: PassengerData) {
  return apiRequest(methodUrl("passenger", "register_passenger"), {
    method: "POST",
    body: JSON.stringify({ passenger_data }),
  });
}

export async function lookupPassengerById(id_number: string) {
  return apiRequest<PassengerRecord | null>(methodUrl("passenger", "lookup_passenger_by_id"), {
    method: "POST",
    body: JSON.stringify({ id_number }),
  });
}

export async function listPassengers(opts?: {
  limit?: number;
  offset?: number;
  search?: string;
}) {
  return apiRequest<PaginatedResponse<PassengerRecord>>(methodUrl("portal", "list_passengers"), {
    method: "POST",
    body: JSON.stringify({
      limit: opts?.limit ?? 50,
      offset: opts?.offset ?? 0,
      search: opts?.search ?? null,
    }),
  });
}
